lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift

```diff
@@ -0,0 +1,461 @@
+import SwiftSyntax
+import SwiftSyntaxMacros
+import SwiftDiagnostics
+import _StringProcessing // for String.contains(_:)
+
+public enum DebugDescriptionMacro {}
+public enum _DebugDescriptionPropertyMacro {}
+
+/// A macro which orchestrates conversion of a description property to an LLDB type summary.
+///
+/// The job of conversion is split across two macros. This macro performs some analysis on the attached
+/// type, and then delegates to `@_DebugDescriptionProperty` to perform the conversion step.
+extension DebugDescriptionMacro: MemberAttributeMacro {
+    public static func expansion<DeclGroup, DeclSyntax, Context>(
+        of node: AttributeSyntax,
+        attachedTo declaration: DeclGroup,
+        providingAttributesFor member: DeclSyntax,
+        in context: Context
+    )
+    throws -> [AttributeSyntax]
+    where DeclGroup: DeclGroupSyntax, DeclSyntax: DeclSyntaxProtocol, Context: MacroExpansionContext
```

Comment:
I will add a comment. The reason it's written this way is because of an lldb bug which doesn't yet support `some` style generics. The way it's written allows me to `po` swift-syntax values.

Reply:
In that case, I would prefer to not use `DeclSyntax` as a generic type because it shadows the `DeclSyntax` type from SwiftSyntax.

Reply:
I updated this PR to use the `some` syntax, and kept my local working copy with the generic syntax.